BundleClient/app/src/main/java/net/discdd/bundleclient/UsbFragment.java

```diff
@@ -115,8 +121,17 @@ private void showUsbAttachedToast() {
         Toast.makeText(getActivity(), "USB device attached", Toast.LENGTH_SHORT).show();
     }
 
+    //Method to check if /DDD_transport directory exists
     private boolean usbDirExists() {
-        // Implement your directory check logic here
+        List<StorageVolume> storageVolumeList = storageManager.getStorageVolumes();
+        for(StorageVolume storageVolume: storageVolumeList) {
+            if(storageVolume.isRemovable()) {
+                File fileUsb = new File(storageVolume.getDirectory().getPath() + usbDirName);
```

Comment:
maybe you can try storageVolume.getDirectory().getPath().resolve(usbDirName).toFile();